#### **Der Datensatz:**

In [127]:
#Importe
import pandas as pd 
pd.set_option('display.max_columns', None)

# Daten Einlesen
df = pd.DataFrame()
df = pd.read_csv('Global YouTube Statistics.csv', encoding='iso-8859-1')

Datenquelle: Die Daten wurden von der Plattform Kaggle bezogen: [Global YouTube Statistics 2023](https://www.kaggle.com/datasets/nelgiriyewithana/global-youtube-statistics-2023/data).

Die Daten stammen von dem Kaggle-Nutzer `Nidula Elgiriyewithana` und sind aus dem Jahr 2023. Laut `Nidula Elgiriyewithana` wurde der Datensatz sorgfältig aus verschiedenen seriösen Quellen zusammengestellt, um die Genauigkeit und Zuverlässigkeit der präsentierten Informationen zu gewährleisten. `Nidula Elgiriyewithana` ist ein aktiver Kaggle-Nutzer, der eine Reihe von hochbewerteten Datensätzen hochgeladen hat und das Abzeichen "Datasets Grandmaster" besitzt. Zudem ist sein LinkedIn-Profil auf Kaggle verlinkt, was bestätigt, dass er als Data Scientist tätig ist und bereits während seines Studiums relevante Erfahrungen gesammelt hat. Diese Faktoren bestätigen seine Seriosität und die Vertrauenswürdigkeit des Datensatzes.

Zusätzlich lässt sich die Korrektheit der Daten leicht überprüfen. Die YouTube-Daten können durch die Website Social Blade validiert werden.

Überprüfungsquelle: [Social Blade - Make Joke Of](https://socialblade.com/youtube/c/makejokeof)

Als Beispiel nehmen wir uns den YouTuber aus der letzten Zeile des Datensatzes:

In [128]:
df.tail(1)

rank      Youtuber  subscribers   video views category         Title  \
994   995  Make Joke Of     12300000  2.129774e+09   Comedy  Make Joke Of   

     uploads Country Abbreviation channel_type  video_views_rank  \
994       62   India           IN       Comedy            4568.0   

     country_rank  channel_type_rank  video_views_for_the_last_30_days  \
994         125.0               44.0                        24000000.0   

     lowest_monthly_earnings  highest_monthly_earnings  \
994                   6000.0                   96000.0   

     lowest_yearly_earnings  highest_yearly_earnings  \
994                 72000.0                1200000.0   

     subscribers_for_last_30_days  created_year created_month  created_date  \
994                      100000.0        2017.0           Aug           1.0   

     Gross tertiary education enrollment (%)    Population  Unemployment rate  \
994                                     28.1  1.366418e+09               5.36   

     Urban_population   Latitude  Longitude  
994       471031528.0  20.593684   78.96288

Auf Social Blade steht dass der Youtuber aktuell (27.06.2024) 12.7 Millionen Abonnenten hat. Bedenkt man, dass die Daten aus dem Jahr 2023 kommen scheinen die Angaben gut zusammenzupassen, denn im Datensatz sind es 12.3 Millionen Abonnenten. Die Kategorie und das Land stimmen auch überein. Die Uploads sind aktuell 75. Im Datensatz steht 62. Das kann man leicht direkt auf dem Kanal des Youtubers prüfen. Die angaben kommen auch so hin genau wie die anderen angaben nach vergleich mit diesem sowie anderen Youtubern. `Nidula` hat die Youtube Daten möglicherweise aus Social Blade gescrapet, da die Spalten Namen mit denen auf `Social Blade` übereinstimmen. Social Blade ist eine seriöse Seite, die auch Youtube zertifiziert ist. 

Die geografischen Werte für longitude and latitude können auch leicht durch Google suche überprüft werden und stimmen mit dem angegebenen Land überein, genau wie andere Datenpunkte. Der Datensatz scheint also wie von dem ersteller behauptet seriös zu sein und somit geeignet für ein wissenschaftliches Journal. 

#### **IDA und Preprocessing:**

In [129]:
print('Der vorhandene Datensatz beschreibt', df.shape[0], 'Youtuber', 'mit', df.shape[1], 'unterschiedlichen Features.')

Der vorhandene Datensatz beschreibt 995 Youtuber mit 28 unterschiedlichen Features.


In [130]:
df.columns

Index(['rank', 'Youtuber', 'subscribers', 'video views', 'category', 'Title',
       'uploads', 'Country', 'Abbreviation', 'channel_type',
       'video_views_rank', 'country_rank', 'channel_type_rank',
       'video_views_for_the_last_30_days', 'lowest_monthly_earnings',
       'highest_monthly_earnings', 'lowest_yearly_earnings',
       'highest_yearly_earnings', 'subscribers_for_last_30_days',
       'created_year', 'created_month', 'created_date',
       'Gross tertiary education enrollment (%)', 'Population',
       'Unemployment rate', 'Urban_population', 'Latitude', 'Longitude'],
      dtype='object')

In [131]:
df.head(5)

rank                    Youtuber  subscribers   video views  \
0     1                    T-Series    245000000  2.280000e+11   
1     2              YouTube Movies    170000000  0.000000e+00   
2     3                     MrBeast    166000000  2.836884e+10   
3     4  Cocomelon - Nursery Rhymes    162000000  1.640000e+11   
4     5                   SET India    159000000  1.480000e+11   

           category                       Title  uploads        Country  \
0             Music                    T-Series    20082          India   
1  Film & Animation               youtubemovies        1  United States   
2     Entertainment                     MrBeast      741  United States   
3         Education  Cocomelon - Nursery Rhymes      966  United States   
4             Shows                   SET India   116536          India   

  Abbreviation   channel_type  video_views_rank  country_rank  \
0           IN          Music               1.0           1.0   
1           US          Games         4055159.0        7670.0   
2           US  Entertainment              48.0           1.0   
3           US      Education               2.0           2.0   
4           IN  Entertainment               3.0           2.0   

   channel_type_rank  video_views_for_the_last_30_days  \
0                1.0                      2.258000e+09   
1             7423.0                      1.200000e+01   
2                1.0                      1.348000e+09   
3                1.0                      1.975000e+09   
4                2.0                      1.824000e+09   

   lowest_monthly_earnings  highest_monthly_earnings  lowest_yearly_earnings  \
0                 564600.0                9000000.00              6800000.00   
1                      0.0                      0.05                    0.04   
2                 337000.0                5400000.00              4000000.00   
3                 493800.0                7900000.00              5900000.00   
4                 455900.0                7300000.00              5500000.00   

   highest_yearly_earnings  subscribers_for_last_30_days  created_year  \
0             1.084000e+08                     2000000.0        2006.0   
1             5.800000e-01                           NaN        2006.0   
2             6.470000e+07                     8000000.0        2012.0   
3             9.480000e+07                     1000000.0        2006.0   
4             8.750000e+07                     1000000.0        2006.0   

  created_month  created_date  Gross tertiary education enrollment (%)  \
0           Mar          13.0                                     28.1   
1           Mar           5.0                                     88.2   
2           Feb          20.0                                     88.2   
3           Sep           1.0                                     88.2   
4           Sep          20.0                                     28.1   

     Population  Unemployment rate  Urban_population   Latitude  Longitude  
0  1.366418e+09               5.36       471031528.0  20.593684  78.962880  
1  3.282395e+08              14.70       270663028.0  37.090240 -95.712891  
2  3.282395e+08              14.70       270663028.0  37.090240 -95.712891  
3  3.282395e+08              14.70       270663028.0  37.090240 -95.712891  
4  1.366418e+09               5.36       471031528.0  20.593684  78.962880

Tabellarische Darstellung der Features des Datensatzes:

| Attribut                                  | Beschreibung                                                                                |
|-------------------------------------------|---------------------------------------------------------------------------------------------|
| `rank`                                    | Position des YouTube-Kanals basierend auf der Anzahl der Abonnenten                         |
| `Youtuber`                                | Name des YouTube-Kanals                                                                     |
| `subscribers`                             | Anzahl der Abonnenten des Kanals                                                            |
| `video views`                             | Gesamtanzahl der Videoaufrufe auf dem Kanal                                                 |
| `category`                                | Kategorie oder Nische des Kanals                                                            |
| `Title`                                   | Titel des YouTube-Kanals                                                                    |
| `uploads`                                 | Gesamtzahl der hochgeladenen Videos auf dem Kanal                                           |
| `Country`                                 | Land, in dem der YouTube-Kanal seinen Ursprung hat                                          |
| `Abbreviation`                            | Abkürzung des Landes                                                                        |
| `channel_type`                            | Art des YouTube-Kanals (z. B. Einzelperson, Marke)                                          |
| `video_views_rank`                        | Rang des Kanals basierend auf der Gesamtzahl der Videoaufrufe                               |
| `country_rank`                            | Rang des Kanals basierend auf der Anzahl der Abonnenten innerhalb seines Landes             |
| `channel_type_rank`                       | Rang des Kanals basierend auf seiner Art (Einzelperson oder Marke)                          |
| `video_views_for_the_last_30_days`        | Gesamtzahl der Videoaufrufe in den letzten 30 Tagen                                         |
| `lowest_monthly_earnings`                 | Niedrigste geschätzte monatliche Einnahmen des Kanals                                       |
| `highest_monthly_earnings`                | Höchste geschätzte monatliche Einnahmen des Kanals                                          |
| `lowest_yearly_earnings`                  | Niedrigste geschätzte jährliche Einnahmen des Kanals                                        |
| `highest_yearly_earnings`                 | Höchste geschätzte jährliche Einnahmen des Kanals                                           |
| `subscribers_for_last_30_days`            | Anzahl der neuen Abonnenten in den letzten 30 Tagen                                         |
| `created_year`                            | Jahr der Erstellung des YouTube-Kanals                                                      |
| `created_month`                           | Monat der Erstellung des YouTube-Kanals                                                     |
| `created_date`                            | Tag der Erstellung des YouTube-Kanals                                                 |
| `Gross tertiary education enrollment (%)` | Prozentsatz der Bevölkerung, die in tertiärer Bildung eingeschrieben ist                    |
| `Population`                              | Gesamtbevölkerung des Landes                                                                |
| `Unemployment rate`                       | Arbeitslosenquote im Land                                                                   |
| `Urban_population`                        | Prozentsatz der Bevölkerung, die in städtischen Gebieten lebt                               |
| `Latitude`                                | Breitengrad der geografischen Lage des Landes                                               |
| `Longitude`                               | Längengrad der geografischen Lage des Landes                                                |

In [132]:
# Datentypen ansehen
df.dtypes

rank                                         int64
Youtuber                                    object
subscribers                                  int64
video views                                float64
category                                    object
Title                                       object
uploads                                      int64
Country                                     object
Abbreviation                                object
channel_type                                object
video_views_rank                           float64
country_rank                               float64
channel_type_rank                          float64
video_views_for_the_last_30_days           float64
lowest_monthly_earnings                    float64
highest_monthly_earnings                   float64
lowest_yearly_earnings                     float64
highest_yearly_earnings                    float64
subscribers_for_last_30_days               float64
created_year                   

Der größte Teil des Datensatzes besteht aus numerischen Werten. Unter diesen befinden sich sieben kategoriale Attribute. Das Attribut `created_month` lässt sich als ordinal klassifizieren, da es eine logische Reihenfolge der Monate repräsentiert. Die Attribute `created_year` und `created_date`, welche derzeit als `float` deklariert sind und exakte Jahres- und Tagesangaben enthalten, werden aus ästhetischen Gründen und zur Klarstellung in ganze Zahlen umgewandelt. Zusätzlich wird das Attribut `created_date` in `created_day` umbenannt, da der Spaltenname irreführend ist. Weitere Attribute, die als `float` deklariert sind, aber nur ganze Zahlen enthalten sollten, wie `Urban_population`, `subscribers_for_last_30_days`, `video_views_for_the_last_30_days`, `channel_type_rank`, `country_rank` und `video_views_rank`, werden ebenfalls in ganze Zahlen umgewandelt, um Genauigkeit und Konsistenz in der Datenrepräsentation zu gewährleisten.

In [133]:
# Konvertieren von 'created_year', 'created_date', und weiteren Attributen von float zu Int64
df['created_year'] = df['created_year'].astype('Int64')
df['created_date'] = df['created_date'].astype('Int64')
df['Urban_population'] = df['Urban_population'].astype('Int64')
df['subscribers_for_last_30_days'] = df['subscribers_for_last_30_days'].astype('Int64')
df['video_views_for_the_last_30_days'] = df['video_views_for_the_last_30_days'].astype('Int64')
df['channel_type_rank'] = df['channel_type_rank'].astype('Int64')
df['country_rank'] = df['country_rank'].astype('Int64')
df['video_views_rank'] = df['video_views_rank'].astype('Int64')

# Umbenennen von 'created_date' zu 'created_day'
df.rename(columns={'created_date': 'created_day'}, inplace=True)

In [134]:
# Datensatz auf Dubletten überprüfen 
print("Anzahl an Dubletten:\n", df.duplicated().sum(), "\n")

Anzahl an Dubletten:
 0 



Keine Dubletten im Datensatz vorhanden, was gut ist da hier nichts weiter gemacht werden muss.

In [135]:
# Zusammenfassung statistischer Kennzeichen
df.describe()

rank   subscribers   video views        uploads  video_views_rank  \
count  995.00000  9.950000e+02  9.950000e+02     995.000000             994.0   
mean   498.00000  2.298241e+07  1.103954e+10    9187.125628     554248.904427   
std    287.37606  1.752611e+07  1.411084e+10   34151.352254    1362782.209881   
min      1.00000  1.230000e+07  0.000000e+00       0.000000               1.0   
25%    249.50000  1.450000e+07  4.288145e+09     194.500000             323.0   
50%    498.00000  1.770000e+07  7.760820e+09     729.000000             915.5   
75%    746.50000  2.460000e+07  1.355470e+10    2667.500000            3584.5   
max    995.00000  2.450000e+08  2.280000e+11  301308.000000         4057944.0   

       country_rank  channel_type_rank  video_views_for_the_last_30_days  \
count         879.0              962.0                             939.0   
mean      386.05347         745.719335                  175610284.794462   
std     1232.244746        1944.386561                  416378166.206619   
min             1.0                1.0                               1.0   
25%            11.0               27.0                        20137500.0   
50%            51.0               65.5                        64085000.0   
75%           123.0             139.75                       168826500.0   
max          7741.0             7741.0                      6589000000.0   

       lowest_monthly_earnings  highest_monthly_earnings  \
count               995.000000              9.950000e+02   
mean              36886.148281              5.898078e+05   
std               71858.724092              1.148622e+06   
min                   0.000000              0.000000e+00   
25%                2700.000000              4.350000e+04   
50%               13300.000000              2.127000e+05   
75%               37900.000000              6.068000e+05   
max              850900.000000              1.360000e+07   

       lowest_yearly_earnings  highest_yearly_earnings  \
count            9.950000e+02             9.950000e+02   
mean             4.422574e+05             7.081814e+06   
std              8.612161e+05             1.379704e+07   
min              0.000000e+00             0.000000e+00   
25%              3.265000e+04             5.217500e+05   
50%              1.595000e+05             2.600000e+06   
75%              4.551000e+05             7.300000e+06   
max              1.020000e+07             1.634000e+08   

       subscribers_for_last_30_days  created_year  created_day  \
count                         658.0         990.0        990.0   
mean                  349079.132219   2012.630303    15.746465   
std                   614355.441025      4.512503      8.77752   
min                             1.0        1970.0          1.0   
25%                        100000.0        2009.0          8.0   
50%                        200000.0        2013.0         16.0   
75%                        400000.0        2016.0         23.0   
max                       8000000.0        2022.0         31.0   

       Gross tertiary education enrollment (%)    Population  \
count                               872.000000  8.720000e+02   
mean                                 63.627752  4.303873e+08   
std                                  26.106893  4.727947e+08   
min                                   7.600000  2.025060e+05   
25%                                  36.300000  8.335541e+07   
50%                                  68.000000  3.282395e+08   
75%                                  88.200000  3.282395e+08   
max                                 113.100000  1.397715e+09   

       Unemployment rate  Urban_population    Latitude   Longitude  
count         872.000000             872.0  872.000000  872.000000  
mean            9.279278  224214981.631881   26.632783  -14.128146  
std             4.888354  154687362.709486   20.560533   84.760809  
min             0.750000           35588.0  -38.416097 -172.104629  
25%             

#### Überprüfung auffälliger Werte:

Einige Werte im Datensatz erfordern eine tiefere Betrachtung, um die Genauigkeit und Verlässlichkeit der Analyse zu gewährleisten:

* Der minimale Wert in der Spalte `created_year` ist 1970. Da YouTube jedoch erst 2005 gegründet wurde, können diese Daten nicht korrekt sein. Ich werde die betroffene Zeile entfernen, nachdem ich überprüft habe, ob es im Datensatz weitere Kanäle gibt, die ein Erstellungsdatum vor 2005 aufweisen.

* Video Views von 0: Es gibt Kanäle mit null Videoaufrufen. Dies könnte entweder auf private Videos, gelöschte Inhalte oder einen Fehler in der Datenerfassung hinweisen. Eine Überprüfung dieser Kanäle ist notwendig, um die Ursachen zu verstehen und die Datenqualität zu sichern.

In [136]:
# Zählen der Zeilen, bei denen 'created_year' vor 2005 liegt
df[df['created_year'] < 2005].shape[0]

1

Es scheint nur einmalig zu sein, was gut ist. Im nächsten Schritt lösche ich diese Zeile.

In [137]:
# Identifizieren der Zeile, bei dem 'created_year' 1970 ist
index_to_drop = df[df['created_year'] == 1970].index

# Entfernen dieser Zeile aus dem DataFrame
df.drop(index_to_drop, inplace=True)

**0 Video Aufrufe Analyse:**

In [138]:
# Zeilen mit 0 Videoaufrufen
df[df['video views'] == 0]

rank            Youtuber  subscribers  video views          category  \
1       2      YouTube Movies    170000000          0.0  Film & Animation   
5       6               Music    119000000          0.0               NaN   
12     13              Gaming     93600000          0.0               NaN   
18     19              Sports     75000000          0.0               NaN   
102   103                News     36300000          0.0               NaN   
173   174  Popular on YouTube     29300000          0.0               NaN   
360   361   Minecraft - Topic     20900000          0.0               NaN   
592   593                Live     16100000          0.0               NaN   

                  Title  uploads        Country Abbreviation   channel_type  \
1         youtubemovies        1  United States           US          Games   
5                 Music        0            NaN          NaN          Music   
12               Gaming        0            NaN          NaN          Games   
18               sports        3  United States           US  Entertainment   
102                News        0            NaN          NaN            NaN   
173  Popular on Youtube        3            NaN          NaN      Education   
360   Minecraft - Topic        0            NaN          NaN          Games   
592                Live        0            NaN          NaN            NaN   

     video_views_rank  country_rank  channel_type_rank  \
1             4055159          7670               7423   
5             4057944          <NA>               <NA>   
12            4057944          <NA>                  1   
18            3898122          6266               5395   
102           4057944          <NA>               <NA>   
173           4019724          <NA>               4847   
360           4057944          <NA>                 24   
592           4057944          <NA>               <NA>   

     video_views_for_the_last_30_days  lowest_monthly_earnings  \
1                                  12                      0.0   
5                                <NA>                      0.0   
12                               <NA>                      0.0   
18                                 16                      0.0   
102                              <NA>                      0.0   
173                                 1                      0.0   
360                              <NA>                      0.0   
592                              <NA>                      0.0   

     highest_monthly_earnings  lowest_yearly_earnings  \
1                        0.05                    0.04   
5                        0.00                    0.00   
12                       0.00                    0.00   
18                       0.06                    0.05   
102                      0.00                    0.00   
173                      0.00                    0.00   
360                      0.00                    0.00   
592                      0.00                    0.00   

     highest_yearly_earnings  subscribers_for_last_30_days  created_year  \
1                       0.58                          <NA>          2006   
5                       0.00                          <NA>          2013   
12                      0.00                          <NA>          2013   
18                      0.77                          <NA>          2006   
102                     0.00                          <NA>          2013   
173                     0.05                          <NA>          2017   
360                     0.00                        300000          2013   
592                     0.00                          <NA>          2015   

    created_month  created_day  Gross tertiary education enrollment (%)  \
1             Mar            5                                     88.2   
5             Sep           24                                      NaN   
12            Dec           15                           

Sieht man sich die Namen der Youtube Kanäle an fällt auf, dass das keine von Nutzern erstelle Kanäle sind, sondern von Youtube automatisch erstelle Playlist Kanäle, die Videos von verschiedenen Youtubern enthalten. Da dieses Journal sich auf die quantitative Analyse von Youtubern fokussiert werden diese Kanäle gelöscht. 

In [139]:
# Liste der zu löschenden Zeilenindizes
indices_to_drop = [1, 5, 12, 18, 102, 173, 360, 592]

# Entfernen der spezifizierten Zeilen aus dem DataFrame
df = df.drop(indices_to_drop)

In [140]:
# Datensatz auf fehlende Werte überprüfen 
print("Missing Values:\n", df.isnull().sum(), "\n")
print("(Rows, Columns):",df.shape)

Missing Values:
 rank                                         0
Youtuber                                     0
subscribers                                  0
video views                                  0
category                                    39
Title                                        0
uploads                                      0
Country                                    116
Abbreviation                               116
channel_type                                28
video_views_rank                             1
country_rank                               110
channel_type_rank                           30
video_views_for_the_last_30_days            51
lowest_monthly_earnings                      0
highest_monthly_earnings                     0
lowest_yearly_earnings                       0
highest_yearly_earnings                      0
subscribers_for_last_30_days               330
created_year                                 5
created_month                              

Es sind einige Fehlwerte da. Um ein aussagekräftiges Journal ohne Biases zu garantieren werde ich alle Spalten einzeln betrachten und eine passende Datensäuberungsmethode anwenden. 

Die `category` und `channel_type` Spalten haben größtenteils die gleichen Bezeichnungen, weshalb die leeren Werte in category durch die Werte in channel_type ersetzt werden und `channel_type` daraufhin gelöscht, da die Informationen redundant sind. 

In [141]:
# Ersetzen der NaN-Werte in 'category' mit den Werten aus 'channel_type'
df['category'] = df['category'].fillna(df['channel_type'])

# Spalte 'channel_type' löschen
df = df.drop('channel_type', axis=1)

In [142]:
# Filtern der Rest-Zeilen, in denen 'category' leer ist
df[df['category'].isna()]

rank      Youtuber  subscribers   video views category         Title  \
445   446  Busy Fun Ltd     18800000  7.762077e+09      NaN  TG MAYANK YT   

     uploads Country Abbreviation  video_views_rank  country_rank  \
445        2     NaN          NaN           4057882          <NA>   

     channel_type_rank  video_views_for_the_last_30_days  \
445               <NA>                              <NA>   

     lowest_monthly_earnings  highest_monthly_earnings  \
445                      0.0                       0.0   

     lowest_yearly_earnings  highest_yearly_earnings  \
445                     0.0                      0.0   

     subscribers_for_last_30_days  created_year created_month  created_day  \
445                          <NA>          2021           Dec           21   

     Gross tertiary education enrollment (%)  Population  Unemployment rate  \
445                                      NaN         NaN                NaN   

     Urban_population  Latitude  Longitude  
445              <NA>       NaN        NaN

Den fehlenden 'category' Wert werde ich durch recherche des Kanals ersetzen: 
###### Quelle: https://socialblade.com/youtube/c/busyfunltd

In [143]:
# Setzen der 'category' für den Youtuber 'Busy Fun Ltd' auf 'Gaming'
df.loc[df['Youtuber'] == 'Busy Fun Ltd', 'category'] = 'Gaming'

**Leere 'created_year' Werte:** 

In [144]:
# Filtern der Zeilen, in denen 'created_year' NaN ist
df[df['created_year'].isna()]

rank               Youtuber  subscribers   video views        category  \
236   237            Chris Brown     25200000  1.552057e+10           Music   
468   469  Good Mythical Morning     18300000  8.798045e+09   Entertainment   
508   509     The Game Theorists     17600000  3.752347e+09          Gaming   
735   736           LEGENDA FUNK     14500000  2.440718e+09           Music   
762   763           Harry Styles     14400000  5.689224e+09  People & Blogs   

                   Title  uploads        Country Abbreviation  \
236           ChrisBrown        0            NaN          NaN   
468  Goodmythicalmorning        0            NaN          NaN   
508     TheGameTheorists        0      Australia           AU   
735          LegendaFUNK        0         Brazil           BR   
762          harrystyles        0  United States           US   

     video_views_rank  country_rank  channel_type_rank  \
236           3612215          <NA>               <NA>   
468           3612215          3305               <NA>   
508           3612215          6095               <NA>   
735              <NA>          <NA>               <NA>   
762           4037956          7574               7340   

     video_views_for_the_last_30_days  lowest_monthly_earnings  \
236                              <NA>                      0.0   
468                              <NA>                      0.0   
508                              <NA>                      0.0   
735                              <NA>                      0.0   
762                              <NA>                      0.0   

     highest_monthly_earnings  lowest_yearly_earnings  \
236                       0.0                     0.0   
468                       0.0                     0.0   
508                       0.0                     0.0   
735                       0.0                     0.0   
762                       0.0                     0.0   

     highest_yearly_earnings  subscribers_for_last_30_days  created_year  \
236                      0.0                          <NA>          <NA>   
468                      0.0                          <NA>          <NA>   
508                      0.0                          <NA>          <NA>   
735                      0.0                          <NA>          <NA>   
762                      0.0                          <NA>          <NA>   

    created_month  created_day  Gross tertiary education enrollment (%)  \
236           NaN         <NA>                                      NaN   
468           NaN         <NA>                                      NaN   
508           NaN         <NA>                                    113.1   
735           NaN         <NA>                                     51.3   
762           NaN         <NA>                                     88.2   

      Population  Unemployment rate  Urban_population   Latitude   Longitude  
236          NaN                NaN              <NA>        NaN         NaN  
468          NaN                NaN              <NA>        NaN         NaN  
508   25766605.0               5.27          21844756 -25.274398  133.775136  
735  212559417.0              12.08         183241641 -14.235004  -51.925280  
762  328239523.0              14.70         270663028  37.090240  -95.712891

Es gibt 5 Kanäle, die keine Informationen zu ihrem Erstellungsdatum haben. Diese werde ich durch Recherche der Kanalinformationen auf *Social Blade* ergänzen. Die restlichen leeren Werte werde ich ebenfalls auffüllen, jedoch mit aktuellen Daten aus dem Jahr 2024. Dadurch können realistische und annähernd genaue Werte gewährleistet werden, anstatt Methoden wie das Füllen durch den Mittelwert zu verwenden.

In [145]:
# Aktualisieren der Werte für den Youtuber 'Good Mythical Morning'
df.loc[df['Youtuber'] == 'Good Mythical Morning', 'Country'] = 'United States'
df.loc[df['Youtuber'] == 'Good Mythical Morning', 'Abbreviation'] = 'US'
df.loc[df['Youtuber'] == 'Good Mythical Morning', 'created_year'] = 2008
df.loc[df['Youtuber'] == 'Good Mythical Morning', 'created_month'] = 'Sep'
df.loc[df['Youtuber'] == 'Good Mythical Morning', 'created_day'] = 17
df.loc[df['Youtuber'] == 'Good Mythical Morning', 'uploads'] = 3194
df.loc[df['Youtuber'] == 'Good Mythical Morning', 'subscribers_for_last_30_days'] = 54493135
df.loc[df['Youtuber'] == 'Good Mythical Morning', 'lowest_yearly_earnings'] = 148800
df.loc[df['Youtuber'] == 'Good Mythical Morning', 'highest_yearly_earnings'] = 2400000
df.loc[df['Youtuber'] == 'Good Mythical Morning', 'lowest_monthly_earnings'] = 12400
df.loc[df['Youtuber'] == 'Good Mythical Morning', 'highest_monthly_earnings'] = 198400
df.loc[df['Youtuber'] == 'Good Mythical Morning', 'video_views_for_the_last_30_days'] = 71171130
df.loc[df['Youtuber'] == 'Good Mythical Morning', 'channel_type_rank'] = 151

# Überprüfen der Änderungen
df[df['Youtuber'] == 'Good Mythical Morning']

rank               Youtuber  subscribers   video views       category  \
468   469  Good Mythical Morning     18300000  8.798045e+09  Entertainment   

                   Title  uploads        Country Abbreviation  \
468  Goodmythicalmorning     3194  United States           US   

     video_views_rank  country_rank  channel_type_rank  \
468           3612215          3305                151   

     video_views_for_the_last_30_days  lowest_monthly_earnings  \
468                          71171130                  12400.0   

     highest_monthly_earnings  lowest_yearly_earnings  \
468                  198400.0                148800.0   

     highest_yearly_earnings  subscribers_for_last_30_days  created_year  \
468                2400000.0                      54493135          2008   

    created_month  created_day  Gross tertiary education enrollment (%)  \
468           Sep           17                                      NaN   

     Population  Unemployment rate  Urban_population  Latitude  Longitude  
468         NaN                NaN              <NA>       NaN        NaN

In [146]:
# Aktualisieren der Werte für den Youtuber 'The Game Theorists'
df.loc[df['Youtuber'] == 'The Game Theorists', 'Country'] = 'United States'
df.loc[df['Youtuber'] == 'The Game Theorists', 'Abbreviation'] = 'US'
df.loc[df['Youtuber'] == 'The Game Theorists', 'created_year'] = 2009
df.loc[df['Youtuber'] == 'The Game Theorists', 'created_month'] = 'Aug'
df.loc[df['Youtuber'] == 'The Game Theorists', 'created_day'] = 22
df.loc[df['Youtuber'] == 'The Game Theorists', 'uploads'] = 708
df.loc[df['Youtuber'] == 'The Game Theorists', 'subscribers_for_last_30_days'] = 100000
df.loc[df['Youtuber'] == 'The Game Theorists', 'lowest_yearly_earnings'] = 69200
df.loc[df['Youtuber'] == 'The Game Theorists', 'highest_yearly_earnings'] = 1100000
df.loc[df['Youtuber'] == 'The Game Theorists', 'lowest_monthly_earnings'] = 5800
df.loc[df['Youtuber'] == 'The Game Theorists', 'highest_monthly_earnings'] = 92300
df.loc[df['Youtuber'] == 'The Game Theorists', 'video_views_for_the_last_30_days'] = 21268433
df.loc[df['Youtuber'] == 'The Game Theorists', 'channel_type_rank'] = 37
df.loc[df['Youtuber'] == 'The Game Theorists', 'country_rank'] = 133

# Überprüfen der Änderungen
df[df['Youtuber'] == 'The Game Theorists']

rank            Youtuber  subscribers   video views category  \
508   509  The Game Theorists     17600000  3.752347e+09   Gaming   

                Title  uploads        Country Abbreviation  video_views_rank  \
508  TheGameTheorists      708  United States           US           3612215   

     country_rank  channel_type_rank  video_views_for_the_last_30_days  \
508           133                 37                          21268433   

     lowest_monthly_earnings  highest_monthly_earnings  \
508                   5800.0                   92300.0   

     lowest_yearly_earnings  highest_yearly_earnings  \
508                 69200.0                1100000.0   

     subscribers_for_last_30_days  created_year created_month  created_day  \
508                        100000          2009           Aug           22   

     Gross tertiary education enrollment (%)  Population  Unemployment rate  \
508                                    113.1  25766605.0               5.27   

     Urban_population   Latitude   Longitude  
508          21844756 -25.274398  133.775136

In [147]:
# Aktualisieren der Werte für den Youtuber 'Chris Brown'
df.loc[df['Youtuber'] == 'Chris Brown', 'Country'] = 'United States'
df.loc[df['Youtuber'] == 'Chris Brown', 'Abbreviation'] = 'US'
df.loc[df['Youtuber'] == 'Chris Brown', 'created_year'] = 2006
df.loc[df['Youtuber'] == 'Chris Brown', 'created_month'] = 'Dec'
df.loc[df['Youtuber'] == 'Chris Brown', 'created_day'] = 20
df.loc[df['Youtuber'] == 'Chris Brown', 'uploads'] = 56
df.loc[df['Youtuber'] == 'Chris Brown', 'subscribers_for_last_30_days'] = 20000
df.loc[df['Youtuber'] == 'Chris Brown', 'lowest_yearly_earnings'] = 475000
df.loc[df['Youtuber'] == 'Chris Brown', 'highest_yearly_earnings'] = 7600000
df.loc[df['Youtuber'] == 'Chris Brown', 'lowest_monthly_earnings'] = 39600
df.loc[df['Youtuber'] == 'Chris Brown', 'highest_monthly_earnings'] = 633300
df.loc[df['Youtuber'] == 'Chris Brown', 'video_views_for_the_last_30_days'] = 194394443
df.loc[df['Youtuber'] == 'Chris Brown', 'channel_type_rank'] = 69
df.loc[df['Youtuber'] == 'Chris Brown', 'country_rank'] = 78

# Überprüfen der Änderungen
df[df['Youtuber'] == 'Chris Brown']

rank     Youtuber  subscribers   video views category       Title  \
236   237  Chris Brown     25200000  1.552057e+10    Music  ChrisBrown   

     uploads        Country Abbreviation  video_views_rank  country_rank  \
236       56  United States           US           3612215            78   

     channel_type_rank  video_views_for_the_last_30_days  \
236                 69                         194394443   

     lowest_monthly_earnings  highest_monthly_earnings  \
236                  39600.0                  633300.0   

     lowest_yearly_earnings  highest_yearly_earnings  \
236                475000.0                7600000.0   

     subscribers_for_last_30_days  created_year created_month  created_day  \
236                         20000          2006           Dec           20   

     Gross tertiary education enrollment (%)  Population  Unemployment rate  \
236                                      NaN         NaN                NaN   

     Urban_population  Latitude  Longitude  
236              <NA>       NaN        NaN

In [148]:
# Aktualisieren der Werte für den Youtuber 'LEGENDA FUNK'
df.loc[df['Youtuber'] == 'LEGENDA FUNK', 'created_year'] = 2013
df.loc[df['Youtuber'] == 'LEGENDA FUNK', 'created_month'] = 'May'
df.loc[df['Youtuber'] == 'LEGENDA FUNK', 'created_day'] = 11
df.loc[df['Youtuber'] == 'LEGENDA FUNK', 'uploads'] = 6100
df.loc[df['Youtuber'] == 'LEGENDA FUNK', 'subscribers_for_last_30_days'] = -24046461
df.loc[df['Youtuber'] == 'LEGENDA FUNK', 'video_views_for_the_last_30_days'] = 3882676
df.loc[df['Youtuber'] == 'LEGENDA FUNK', 'channel_type_rank'] = 140
df.loc[df['Youtuber'] == 'LEGENDA FUNK', 'country_rank'] = 46
df.loc[df['Youtuber'] == 'LEGENDA FUNK', 'video_views_rank'] = 5536

# Überprüfen der Änderungen
df[df['Youtuber'] == 'LEGENDA FUNK']

rank      Youtuber  subscribers   video views category        Title  \
735   736  LEGENDA FUNK     14500000  2.440718e+09    Music  LegendaFUNK   

     uploads Country Abbreviation  video_views_rank  country_rank  \
735     6100  Brazil           BR              5536            46   

     channel_type_rank  video_views_for_the_last_30_days  \
735                140                           3882676   

     lowest_monthly_earnings  highest_monthly_earnings  \
735                      0.0                       0.0   

     lowest_yearly_earnings  highest_yearly_earnings  \
735                     0.0                      0.0   

     subscribers_for_last_30_days  created_year created_month  created_day  \
735                     -24046461          2013           May           11   

     Gross tertiary education enrollment (%)   Population  Unemployment rate  \
735                                     51.3  212559417.0              12.08   

     Urban_population   Latitude  Longitude  
735         183241641 -14.235004  -51.92528

In [149]:
# Aktualisieren der Werte für den Youtuber 'Harry Styles'
df.loc[df['Youtuber'] == 'Harry Styles', 'created_year'] = 2017
df.loc[df['Youtuber'] == 'Harry Styles', 'created_month'] = 'Mar'
df.loc[df['Youtuber'] == 'Harry Styles', 'created_day'] = 8
df.loc[df['Youtuber'] == 'Harry Styles', 'uploads'] = 17
df.loc[df['Youtuber'] == 'Harry Styles', 'subscribers_for_last_30_days'] = 100000
df.loc[df['Youtuber'] == 'Harry Styles', 'lowest_yearly_earnings'] = 178800
df.loc[df['Youtuber'] == 'Harry Styles', 'highest_yearly_earnings'] = 2900000
df.loc[df['Youtuber'] == 'Harry Styles', 'lowest_monthly_earnings'] = 14900
df.loc[df['Youtuber'] == 'Harry Styles', 'highest_monthly_earnings'] = 238300
df.loc[df['Youtuber'] == 'Harry Styles', 'video_views_for_the_last_30_days'] = 113745926
df.loc[df['Youtuber'] == 'Harry Styles', 'channel_type_rank'] = 138

# Überprüfen der Änderungen
df[df['Youtuber'] == 'Harry Styles']

rank      Youtuber  subscribers   video views        category  \
762   763  Harry Styles     14400000  5.689224e+09  People & Blogs   

           Title  uploads        Country Abbreviation  video_views_rank  \
762  harrystyles       17  United States           US           4037956   

     country_rank  channel_type_rank  video_views_for_the_last_30_days  \
762          7574                138                         113745926   

     lowest_monthly_earnings  highest_monthly_earnings  \
762                  14900.0                  238300.0   

     lowest_yearly_earnings  highest_yearly_earnings  \
762                178800.0                2900000.0   

     subscribers_for_last_30_days  created_year created_month  created_day  \
762                        100000          2017           Mar            8   

     Gross tertiary education enrollment (%)   Population  Unemployment rate  \
762                                     88.2  328239523.0               14.7   

     Urban_population  Latitude  Longitude  
762         270663028  37.09024 -95.712891

**Im nächsten Schritt überprüfe ich ob alle Youtuber Namen richtig eingelesen wurden:**

In [150]:
# Prüfen auf nicht-ASCII-Zeichen in der 'Youtuber' Spalte
df[df['Youtuber'].apply(lambda x: not x.isascii())]

rank                                       Youtuber  subscribers  \
6       7                            ýýý Kids Diana Show    112000000   
55     56                                   Voc���     44700000   
64     65                     ýýýýýýýý ýý ýýýýýýýýýýýýýý     43200000   
76     77                                 shfa2 - ��     39700000   
91     92                                  Vlad vï¿½ï¿½ï     37900000   
..    ...                                            ...          ...   
954   955                    Little Baby Bum en Espaï¿½ï     12500000   
956   957                                  M����     12500000   
975   976                              Gustavo Par��     12400000   
979   980  DaniRep | +6 Vï¿½ï¿½ï¿½ï¿½ï¿½ï¿½ï¿½ï¿½ï¿½ï¿½ï     12400000   
990   991                                  Natan por Aï¿     12300000   

      video views          category  \
6    9.324704e+10    People & Blogs   
55   7.828611e+09     Entertainment   
64   3.645873e+10  Film & Animation   
76   2.388482e+10    People & Blogs   
91   2.351015e+10     Entertainment   
..            ...               ...   
954  6.956320e+09         Education   
956  3.140883e+09   Travel & Events   
975  2.602614e+09            Comedy   
979  6.933661e+09            Gaming   
990  9.029610e+09            Sports   

                                             Title  uploads  \
6                              ýýý Kids Diana Show     1111   
55                                    Voc���     1558   
64                      ýýýýýýýý ýý ýýýýýýýýýýýýýý     1478   
76                                  shfa2 - ��     1596   
91                                   Vlad vï¿½ï¿½ï      515   
..                                             ...      ...   
954                    Little Baby Bum en Espaï¿½ï     1459   
956                                  M����      766   
975                                GustavoParodias        9   
979  DaniRep | +6 Vï¿½ï¿½ï¿½ï¿½ï¿½ï¿½ï¿½ï¿½ï¿½ï¿½ï    12419   
990                                  Natan por Aï¿     1200   

                  Country Abbreviation  video_views_rank  country_rank  \
6           United States           US                 5             3   
55                 Brazil           BR               681             3   
64                 Russia           RU                26             2   
76   United Arab Emirates           AE                81             1   
91          United States           US                84            28   
..                    ...          ...               ...           ...   
954                 Spain           ES               843            19   
956                Mexico           MX              2691            33   
975                Brazil           BR           4050768          5075   
979                 Spain           ES               847            19   
990                Brazil           BR               525            55   

     channel_type_rank  video_views_for_the_last_30_days  \
6                    3                         731674000   
55                  15                          48032000   
64                   6                         303780000   
76                   2                         247731000   
91                  27                         244093000   
..                 ...                               ...   
954                 46                          16319000   
956                171                          31007000   
975               4894                              <NA>   
979                 68                          82648000   
990                172                         552513000   

     lowest_monthly_earnings  highest_monthly_earnings  \
6                   182900.0                 2900000.0   
55                   12000.0                  192100.0   
64                   75900.0                 1200000.0   
76                   61900.0                  990900.0   
91

Youtube-Kanäle mit Sonderzeichen, die außerhalb des ASCII-Zeichensatzes liegen, wurden mit ungewöhnlichen Zeichen interpretiert. Diese werde ich im nächsten Schritt bereinigen.

In [151]:
# Entfernen von nicht-ASCII-Zeichen aus der 'Youtuber' Spalte
df['Youtuber'] = df['Youtuber'].apply(lambda x: x.encode('ascii', 'ignore').decode('ascii'))

###### Quelle: https://community.databricks.com/t5/data-engineering/removing-non-ascii-and-special-character-in-pyspark/td-p/27782

In [152]:
# Überprüfung ob es noch Kanäle gibt mit Zeichen außerhalb der ASCII-Zeichen
df[df['Youtuber'].apply(lambda x: not x.isascii())]

Empty DataFrame
Columns: [rank, Youtuber, subscribers, video views, category, Title, uploads, Country, Abbreviation, video_views_rank, country_rank, channel_type_rank, video_views_for_the_last_30_days, lowest_monthly_earnings, highest_monthly_earnings, lowest_yearly_earnings, highest_yearly_earnings, subscribers_for_last_30_days, created_year, created_month, created_day, Gross tertiary education enrollment (%), Population, Unemployment rate, Urban_population, Latitude, Longitude]
Index: []

In [153]:
# Zählen leerer Einträge nach der Bereinigung
empty_entries = df[df['Youtuber'].apply(lambda x: x.strip() == '')]
print(f"Anzahl leerer Einträge nach der Bereinigung: {len(empty_entries)}")

Anzahl leerer Einträge nach der Bereinigung: 16


In [154]:
empty_entries

rank Youtuber  subscribers   video views          category  \
64     65              43200000  3.645873e+10  Film & Animation   
161   162              30400000  1.799996e+10     Entertainment   
433   434              19000000  4.924054e+09         Education   
447   448              18800000  9.594189e+09     Entertainment   
561   562              16500000  2.440934e+09     Entertainment   
606   607              15900000  1.845330e+09    People & Blogs   
632   633              15500000  8.265130e+09             Shows   
700   701              14900000  4.390980e+05    People & Blogs   
707   708              14800000  7.018015e+09     Howto & Style   
714   715              14700000  2.230986e+09     Entertainment   
752   753              14400000  6.001543e+08     Entertainment   
784   785              14100000  3.920221e+09     Entertainment   
795   796              14000000  7.719743e+09            Gaming   
810   811              13900000  8.451755e+09            People   
866   867              13300000  6.482687e+09    People & Blogs   
920   921              12900000  5.585085e+09             Shows   

                                         Title  uploads        Country  \
64                  ýýýýýýýý ýý ýýýýýýýýýýýýýý     1478         Russia   
161  ýýýýýýýýýýýýýýý ýýýýýý ýýýýýýýýýýýýýýýýýý      532  United States   
433                        ýýýýýýýýýýýýýý ýýýý       60  United States   
447                     ýýýýýýýýýýýýýýýýýýýýýý      530        Ukraine   
561            ýýýýýýýý ýýýýýýýýýýýýýýýýýýýýýý      421  United States   
606                              Kung Fu Padla        0         Russia   
632                       ýýýýýýýý ýý ýýýýýýýý     1021   Saudi Arabia   
700                                     ýýýýýý        1         Russia   
707                          ýýýýýýýý ýýýýýýýý     2387        Ukraine   
714                    ýýýýýýýýýý ýýýýýýýýýýýý     1385   Saudi Arabia   
752                      ýýýýýýýýýýýýýýýýýýýýý      364    South Korea   
784  ýýýýýýýýýýýýýýýýýýýý ýýýýýýýýýýýýýýýýýýýý       65            NaN   
795                               ýýýýýýýýýýýý     2210         Russia   
810                   ýýýýýýýý ýý ýýýýýýýýýýýý      504  United States   
866                         ýýýýýýýýýýýýýýýýýý      608         Russia   
920               ýýýýýýýý ýýýý ýýýýýýýýýýýýýý     1255        Ukraine   

    Abbreviation  video_views_rank  country_rank  channel_type_rank  \
64            RU                26             2                  6   
161           US               147            46                 46   
433           US              1399           116                111   
447           UA               483             2                113   
561           US              3778           137                133   
606           RU           4057944          3309               5665   
632           SA               619             4                 33   
700           RU           3609784            10                 19   
707           UA               828             3                 25   
714           SA              4276             5                149   
752           KR             21132            14                151   
784          NaN           3999155          <NA>               <NA>   
795           RU               703            12                 56   
810           US               600           161                 50   
866           RU               927            13                163   
920           UA              1164             6                166   

     video_views_for_the_last_30_days  lowest_monthly_earnings  \
64                          303780000                  75900.0   
161                         122926000                  30700.0   
433                          21587000                   5400.0   
447                          68006000                  17000.0   
561                          21953000                   5500.0   
60

Die bereinigten Youtuber Namen sind nun leere Namen. Diesen 16 Youtubern gebe ich den Namen `unknown_youtuber`

In [155]:
# Entfernen von führenden und nachfolgenden Leerzeichen
df['Youtuber'] = df['Youtuber'].str.strip()

df.loc[df['Youtuber'] == '', 'Youtuber'] = 'unknown_youtuber'

# Überprüfung leerer Einträge
empty_entries = df[df['Youtuber'].apply(lambda x: x.strip() == '')]
print(f"Anzahl leerer Einträge: {len(empty_entries)}")

Anzahl leerer Einträge: 0


**Latitude und Longitude Untersuchung:**

In [156]:
# Filtern der Zeilen, bei denen Latitude und Longitude leer sind, aber Country gefüllt ist
df[(df['Latitude'].isnull() | df['Longitude'].isnull()) & df['Country'].notnull()]

rank               Youtuber  subscribers   video views       category  \
236   237            Chris Brown     25200000  1.552057e+10          Music   
468   469  Good Mythical Morning     18300000  8.798045e+09  Entertainment   
663   664                  Musas     15100000  2.400038e+09  Howto & Style   

                   Title  uploads        Country Abbreviation  \
236           ChrisBrown       56  United States           US   
468  Goodmythicalmorning     3194  United States           US   
663           Jordi Sala      159        Andorra           AD   

     video_views_rank  country_rank  channel_type_rank  \
236           3612215            78                 69   
468           3612215          3305                151   
663           3889418           504               7615   

     video_views_for_the_last_30_days  lowest_monthly_earnings  \
236                         194394443                 39600.00   
468                          71171130                 12400.00   
663                                86                     0.02   

     highest_monthly_earnings  lowest_yearly_earnings  \
236                 633300.00               475000.00   
468                 198400.00               148800.00   
663                      0.34                    0.26   

     highest_yearly_earnings  subscribers_for_last_30_days  created_year  \
236                7600000.0                         20000          2006   
468                2400000.0                      54493135          2008   
663                      4.0                          <NA>          2006   

    created_month  created_day  Gross tertiary education enrollment (%)  \
236           Dec           20                                      NaN   
468           Sep           17                                      NaN   
663           Mar           12                                      NaN   

     Population  Unemployment rate  Urban_population  Latitude  Longitude  
236         NaN                NaN              <NA>       NaN        NaN  
468         NaN                NaN              <NA>       NaN        NaN  
663         NaN                NaN              <NA>       NaN        NaN

In [157]:
# Filtern der Zeilen, bei denen Country gleich 'United States' ist
df[df['Country'] == 'United States']

rank                    Youtuber  subscribers   video views  \
2       3                     MrBeast    166000000  2.836884e+10   
3       4  Cocomelon - Nursery Rhymes    162000000  1.640000e+11   
6       7             Kids Diana Show    112000000  9.324704e+10   
9      10               Vlad and Niki     98900000  7.718017e+10   
11     12                         WWE     96000000  7.742847e+10   
..    ...                         ...          ...           ...   
973   974                        LOUD     12400000  2.394143e+09   
983   984                    MoniLina     12400000  1.608681e+10   
987   988               Avril Lavigne     12400000  6.202090e+09   
988   989                      Caylus     12400000  4.779140e+09   
989   990                   Migos ATL     12400000  6.993406e+09   

           category                       Title  uploads        Country  \
2     Entertainment                     MrBeast      741  United States   
3         Education  Cocomelon - Nursery Rhymes      966  United States   
6    People & Blogs         ýýý Kids Diana Show     1111  United States   
9     Entertainment               Vlad and Niki      574  United States   
11           Sports                         WWE    70127  United States   
..              ...                         ...      ...            ...   
973          Gaming                   WatchLOUD      690  United States   
983          Comedy              MoniLinaFamily        0  United States   
987           Music               Avril Lavigne      205  United States   
988       Education                      Caylus     1340  United States   
989           Music                   Migos ATL       99  United States   

    Abbreviation  video_views_rank  country_rank  channel_type_rank  \
2             US                48             1                  1   
3             US                 2             2                  1   
6             US                 5             3                  3   
9             US                 8             5                  6   
11            US                 7             6                  1   
..           ...               ...           ...                ...   
973           US            186431          1795               1759   
983           US           4057944          2774               2499   
987           US               999           176                150   
988           US              1442           175                 68   
989           US               833           175                171   

     video_views_for_the_last_30_days  lowest_monthly_earnings  \
2                          1348000000                 337000.0   
3                          1975000000                 493800.0   
6                           731674000                 182900.0   
9                           580574000                 145100.0   
11                          714614000                 178700.0   
..                                ...                      ...   
973                             27596                      7.0   
983                              <NA>                      0.0   
987                          50188000                  12500.0   
988                         177600000                  44400.0   
989                          49412000                  12400.0   

     highest_monthly_earnings  lowest_yearly_earnings  \
2                   5400000.0               4000000.0   
3                   7900000.0               5900000.0   
6                   2900000.0               2200000.0   
9                   2300000.0               1700000.0   
11                  2900000.0               2100000.0   
..                        ...                     ...   
973                     110.0                    83.0   
983                       0.0                     0.0   
987                  200800.0                150600.0   
988                  710400.0                532800.0   
989           

#### Datenaktualisierung für Youtuber:

Bei der Analyse der geografischen Daten im Datensatz fällt auf, dass alle Youtuber aus den Vereinigten Staaten identische Werte für die Latitude- und Longitude-Werte aufweisen. Aufgrund dieser Konsistenz werden die geografischen Daten für die Youtuber `Chris Brown` und `Good Mythical Morning` aus den Vereinigten Staaten mit diesen Standardwerten aktualisiert:

- **Latitude:** 37.09024
- **Longitude:** -95.712891

Zusätzlich werden für diese Youtuber die folgenden demografischen und geografischen Daten aus den USA übernommen, da diese konsequent gleichbleibend sind:

- **Gross tertiary education enrollment (%):** 88.2
- **Population:** 328,239,523
- **Unemployment rate:** 14.7
- **Urban_population:** 270,663,028

Für den Youtuber `Musas` aus Andorra, der einzigartig im Datensatz vertreten ist, werden die geografischen Daten basierend auf meiner Recherche aktualisiert:

- **Latitude:** 42.506317 
- **Longitude:** 1.521835 
- **Urban_population:** 68,043
- **Unemployment rate:** 1.20
- **Population:** 80,088

###### Quellen: https://www.worldometers.info/world-population/andorra-population/ & https://www.theglobaleconomy.com/Andorra/unemployment_rate_monthly/

In [158]:
# Festlegen der Werte für US Spalten
values_to_set = {
    'Gross tertiary education enrollment (%)': 88.2,
    'Population': 328239523.0,
    'Unemployment rate': 14.7,
    'Urban_population': 270663028,
    'Latitude': 37.09024,
    'Longitude': -95.712891
}

# Aktualisieren der Werte für Chris Brown und Good Mythical Morning
df.loc[df['Youtuber'].isin(['Chris Brown', 'Good Mythical Morning']), list(values_to_set.keys())] = list(values_to_set.values())

# Überprüfen der Änderungen
df[df['Youtuber'].isin(['Chris Brown', 'Good Mythical Morning'])]

rank               Youtuber  subscribers   video views       category  \
236   237            Chris Brown     25200000  1.552057e+10          Music   
468   469  Good Mythical Morning     18300000  8.798045e+09  Entertainment   

                   Title  uploads        Country Abbreviation  \
236           ChrisBrown       56  United States           US   
468  Goodmythicalmorning     3194  United States           US   

     video_views_rank  country_rank  channel_type_rank  \
236           3612215            78                 69   
468           3612215          3305                151   

     video_views_for_the_last_30_days  lowest_monthly_earnings  \
236                         194394443                  39600.0   
468                          71171130                  12400.0   

     highest_monthly_earnings  lowest_yearly_earnings  \
236                  633300.0                475000.0   
468                  198400.0                148800.0   

     highest_yearly_earnings  subscribers_for_last_30_days  created_year  \
236                7600000.0                         20000          2006   
468                2400000.0                      54493135          2008   

    created_month  created_day  Gross tertiary education enrollment (%)  \
236           Dec           20                                     88.2   
468           Sep           17                                     88.2   

      Population  Unemployment rate  Urban_population  Latitude  Longitude  
236  328239523.0               14.7         270663028  37.09024 -95.712891  
468  328239523.0               14.7         270663028  37.09024 -95.712891

In [159]:
# Latitude, Longitude, Urban_population, Unemployment rate und Population für Musas aus Andorra setzen
df.loc[df['Youtuber'] == 'Musas', 'Latitude'] = 42.506317
df.loc[df['Youtuber'] == 'Musas', 'Longitude'] = 1.521835
df.loc[df['Youtuber'] == 'Musas', 'Urban_population'] = 68043
df.loc[df['Youtuber'] == 'Musas', 'Unemployment rate'] = 1.20
df.loc[df['Youtuber'] == 'Musas', 'Population'] = 80088

# Überprüfen der Änderungen
df[df['Youtuber'].isin(['Musas'])]

rank Youtuber  subscribers   video views       category       Title  \
663   664    Musas     15100000  2.400038e+09  Howto & Style  Jordi Sala   

     uploads  Country Abbreviation  video_views_rank  country_rank  \
663      159  Andorra           AD           3889418           504   

     channel_type_rank  video_views_for_the_last_30_days  \
663               7615                                86   

     lowest_monthly_earnings  highest_monthly_earnings  \
663                     0.02                      0.34   

     lowest_yearly_earnings  highest_yearly_earnings  \
663                    0.26                      4.0   

     subscribers_for_last_30_days  created_year created_month  created_day  \
663                          <NA>          2006           Mar           12   

     Gross tertiary education enrollment (%)  Population  Unemployment rate  \
663                                      NaN     80088.0                1.2   

     Urban_population   Latitude  Longitude  
663             68043  42.506317   1.521835

In [160]:
# Datensatz erneut auf fehlende Werte überprüfen 
print("Missing Values:\n", df.isnull().sum(), "\n")
print("(Rows, Columns):",df.shape)

Missing Values:
 rank                                         0
Youtuber                                     0
subscribers                                  0
video views                                  0
category                                     0
Title                                        0
uploads                                      0
Country                                    114
Abbreviation                               114
video_views_rank                             0
country_rank                               108
channel_type_rank                           26
video_views_for_the_last_30_days            46
lowest_monthly_earnings                      0
highest_monthly_earnings                     0
lowest_yearly_earnings                       0
highest_yearly_earnings                      0
subscribers_for_last_30_days               325
created_year                                 0
created_month                                0
created_day                                

#### Behandlung von fehlenden Werten:

Es gibt immer noch einige fehlende Werte in den Spalten. Diese fehlenden Werte werden in den numerischen Spalten als `NaN` oder `pd.NA` belassen, um die Genauigkeit der Originaldaten zu wahren. In den Visualisierungen, die ich mit dem Werkzeugen Plotly durchführe, werden diese fehlenden Werte automatisch ignoriert. Dies gewährleistet, dass die Analysen und Visualisierungen nur auf verfügbaren, validen Daten basieren, ohne dass durch das Ersetzen von fehlenden Werten zusätzliche Verzerrungen eingeführt werden. Mit dieser Methode gewährleiste ich eine transparente und genaue Darstellung der Daten.

#### Umbenennung der Spaltenamen, um den Code sauberer und besser lesbar zu machen: 

In [161]:
df.rename(columns={
    'Youtuber': 'channel_name',
    'video views': 'total_views',
    'Country': 'country',
    'Title': 'title',
    'Abbreviation': 'country_abbr',
    'channel_type': 'channel_type',
    'video_views_rank': 'views_rank',
    'country_rank': 'country_sub_rank',
    'channel_type_rank': 'type_rank',
    'video_views_for_the_last_30_days': 'views_30d',
    'lowest_monthly_earnings': 'min_monthly_earnings',
    'highest_monthly_earnings': 'max_monthly_earnings',
    'lowest_yearly_earnings': 'min_yearly_earnings',
    'highest_yearly_earnings': 'max_yearly_earnings',
    'subscribers_for_last_30_days': 'subs_30d',
    'created_year': 'created_year',
    'created_month': 'created_month',
    'created_date': 'created_day',
    'Gross tertiary education enrollment (%)': 'tertiary_edu_rate',
    'Population': 'population',
    'Unemployment rate': 'unemployment_rate',
    'Urban_population': 'urban_pop',
    'Latitude': 'lat',
    'Longitude': 'long'
}, inplace=True)

In [162]:
df.head(2)

rank channel_name  subscribers   total_views       category     title  \
0     1     T-Series    245000000  2.280000e+11          Music  T-Series   
2     3      MrBeast    166000000  2.836884e+10  Entertainment   MrBeast   

   uploads        country country_abbr  views_rank  country_sub_rank  \
0    20082          India           IN           1                 1   
2      741  United States           US          48                 1   

   type_rank   views_30d  min_monthly_earnings  max_monthly_earnings  \
0          1  2258000000              564600.0             9000000.0   
2          1  1348000000              337000.0             5400000.0   

   min_yearly_earnings  max_yearly_earnings  subs_30d  created_year  \
0            6800000.0          108400000.0   2000000          2006   
2            4000000.0           64700000.0   8000000          2012   

  created_month  created_day  tertiary_edu_rate    population  \
0           Mar           13               28.1  1.366418e+09   
2           Feb           20               88.2  3.282395e+08   

   unemployment_rate  urban_pop        lat       long  
0               5.36  471031528  20.593684  78.962880  
2              14.70  270663028  37.090240 -95.712891

#### Entfernen redundanter Spalten:
* Die Spalte `title`, die dieselben Informationen wie die Spalte `channel_name` enthält, wird entfernt. Dies vermeidet Redundanz und minimiert die Datenmenge, zum Verwalten und Analysieren. 
* Die aktuelle `rank`-Spalte, die den Rang des YouTube-Kanals basierend auf der Anzahl der Abonnenten angibt, ist durch das vorherige Löschen von Zeilen nicht mehr konsistent. Um eine präzise Rangordnung zu gewährleisten, werde ich die aktuelle `rank-`Spalte entfernen, den DataFrame nach den Abonnentenzahlen neu sortieren und den Index, sowie `rank` daraufhin neu setzen.

In [163]:
# Entfernen der aktuellen rank-Spalte
df.drop(columns=['rank'], inplace=True)
df.drop(columns=['title'], inplace=True)

# Neuordnung des DataFrames
df.sort_values(by='subscribers', ascending=False, inplace=True)

# Neusetzen des Indexes und Rang
df.reset_index(drop=True, inplace=True)
df['rank'] = df.index + 1

#### Datumsspalte erstellen falls eine Zeitreihenanalyse gemacht wird:

In [164]:
# Umwandlung von Monatsabkürzungen in Zahlen für die Datumsfunktion
month_mapping = {
    'Jan': '01', 'Feb': '02', 'Mar': '03', 'Apr': '04', 'May': '05', 'Jun': '06',
    'Jul': '07', 'Aug': '08', 'Sep': '09', 'Oct': '10', 'Nov': '11', 'Dec': '12'
}

# Erstellen einer temporären Spalte für die Umwandlung
df['temp_month'] = df['created_month'].map(month_mapping)

# Datumsspalte erstellen
temp_df = df[['created_year', 'temp_month', 'created_day']].rename(columns={
    'created_year': 'year',
    'temp_month': 'month',
    'created_day': 'day'
})
df['date'] = pd.to_datetime(temp_df[['year', 'month', 'day']].astype(str))

# Temporäre Spalte entfernen
df.drop('temp_month', axis=1, inplace=True)

# Ergebnis anzeigen
df[['created_year', 'created_month', 'created_day', 'date']]

created_year created_month  created_day       date
0            2006           Mar           13 2006-03-13
1            2012           Feb           20 2012-02-20
2            2006           Sep            1 2006-09-01
3            2006           Sep           20 2006-09-20
4            2015           May           12 2015-05-12
..            ...           ...          ...        ...
981          2017           Feb           12 2017-02-12
982          2018           Sep           14 2018-09-14
983          2006           Sep           11 2006-09-11
984          2012           May            9 2012-05-09
985          2017           Aug            1 2017-08-01

[986 rows x 4 columns]

In [165]:
# Umbenennen der Kategorien
df['category'] = df['category'].replace({
    'People': 'People & Blogs',
    'Games': 'Gaming',
    'Howto': 'Howto & Style'
})

# Überprüfen, ob die Umbenennung funktioniert hat
print(df['category'].unique())

['Music' 'Entertainment' 'Education' 'Shows' 'People & Blogs' 'Gaming'
 'Sports' 'Film & Animation' 'Howto & Style' 'News & Politics' 'Comedy'
 'Trailers' 'Nonprofits & Activism' 'Science & Technology' 'Movies'
 'Pets & Animals' 'Autos & Vehicles' 'Tech' 'Film' 'Travel & Events']


In dem Story-Notebook ist mir aufgefallen, dass einige Kategorien doppelt vorhanden sind, jedoch mit leicht abweichenden Namensgebungen. Um die Analyse zu vereinfachen und die Konsistenz zu gewährleisten, habe ich diese Kategorien zusammengeführt. Dadurch werden beispielsweise die Kategorien "People" und "People & Blogs" unter "People & Blogs" zusammengefasst.

In [166]:
# Ersetzen von 'United States' durch 'United States of America'
df['country'] = df['country'].replace('United States', 'United States of America')

In [167]:
# Überprüfen, ob die Umbenennung funktioniert hat
print(df['country'].unique())

['India' 'United States of America' 'Japan' 'Russia' 'South Korea' nan
 'United Kingdom' 'Canada' 'Brazil' 'Argentina' 'Chile' 'Cuba'
 'El Salvador' 'Pakistan' 'Philippines' 'Thailand' 'Colombia' 'Barbados'
 'Mexico' 'United Arab Emirates' 'Spain' 'Saudi Arabia' 'Indonesia'
 'Turkey' 'Venezuela' 'Kuwait' 'Jordan' 'Netherlands' 'Australia'
 'Singapore' 'Italy' 'Germany' 'France' 'Sweden' 'Ukraine' 'Afghanistan'
 'Latvia' 'Switzerland' 'Vietnam' 'Malaysia' 'China' 'Iraq' 'Egypt'
 'Andorra' 'Ecuador' 'Morocco' 'Peru' 'Bangladesh' 'Finland' 'Samoa']


Geo Daten für Amerika sind normalerweise unter "United States of America" statt "United States" definiert. Die `country` Spalte enthält jedoch die Bezeichnung "United States", weshalb dies in "United States of America" umgeändert wurde, um eine korrekte Darstellung in Kartenvisualisierungen zu erhalten. 

In [168]:
# Ersetzen von 'Tech' durch 'Science & Technology'
df['category'] = df['category'].replace('Tech', 'Science & Technology')

`Tech` kommt nur einmal vor in der `category` Spalte und gehört zur 'Science & Technology' Kategorie.

#### **Speicherung des bereinigten Datensatzes:**

In [169]:
df.to_csv('Global_YouTube_Statistics_cleaned.csv', index=False)